File: xoom-eventjournal/src/main/java/io/vlingo/xoom/examples/eventjournal/interest/NoopConfigurationInterest.java
package io.vlingo.xoom.examples.eventjournal.interest;

import java.sql.Connection;

import io.vlingo.xoom.symbio.store.common.jdbc.Configuration;

public class NoopConfigurationInterest implements Configuration.ConfigurationInterest {
    @Override
    public void afterConnect(Connection connection) throws Exception {

    }

    @Override
    public void beforeConnect(Configuration configuration) throws Exception {

    }

    @Override
    public void createDatabase(Connection connection, String databaseName) throws Exception {

    }

    @Override
    public void dropDatabase(Connection connection, String databaseName) throws Exception {

    }
}

File: vlingo-iot-android/backend/src/main/kotlin/io/vlingo/hike/infrastructure/NoopConfigurationInterest.kt
package io.vlingo.hike.infrastructure

import io.vlingo.symbio.store.common.jdbc.Configuration
import java.sql.Connection

object NoopConfigurationInterest: Configuration.ConfigurationInterest {
    override fun afterConnect(connection: Connection?) {
    }

    override fun beforeConnect(configuration: Configuration?) {
    }

    override fun dropDatabase(connection: Connection?, databaseName: String?) {
    }

    override fun createDatabase(connection: Connection?, databaseName: String?) {
    }
}